pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceTransaction.java

```diff
@@ -0,0 +1,613 @@
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.LongAdder;
+import org.HdrHistogram.Histogram;
+import org.HdrHistogram.HistogramLogWriter;
+import org.HdrHistogram.Recorder;
+import org.apache.commons.lang3.RandomUtils;
+import org.apache.curator.shaded.com.google.common.util.concurrent.RateLimiter;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.admin.PulsarAdminBuilder;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.ConsumerBuilder;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.ProducerBuilder;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.apache.pulsar.client.api.transaction.Transaction;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.common.partition.PartitionedTopicMetadata;
+import org.apache.pulsar.testclient.utils.PaddingDecimalFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PerformanceTransaction {
+
+
+    private static final LongAdder totalNumTransaction = new LongAdder();
+    private static final LongAdder numTransaction = new LongAdder();
+
+    private static final LongAdder totalMessageAck = new LongAdder();
+    private static final LongAdder messageAck = new LongAdder();
+
+
+    private static Recorder messageAckRecorder = new Recorder(TimeUnit.SECONDS.toMicros(120000), 5);
+    private static Recorder messageAckCumulativeRecorder = new Recorder(TimeUnit.SECONDS.toMicros(120000), 5);
+
+    private static Recorder messageSendRecorder = new Recorder(TimeUnit.SECONDS.toMicros(120000), 5);
+    private static Recorder messageSendRCumulativeRecorder = new Recorder(TimeUnit.SECONDS.toMicros(120000), 5);
+
+
+    @Parameters(commandDescription = "Test pulsar transaction performance.")
+    static class Arguments {
+
+        @Parameter(names = {"-h", "--help"}, description = "Help message", help = true)
+        boolean help;
+
+        @Parameter(names = {"--conf-file"}, description = "Configuration file")
+        public String confFile;
+
+        @Parameter(names = "--topic-c", description = "consumer will be created to consumer this topic", required =
+                true)
+        public List<String> consumerTopic = Collections.singletonList("sub");
+
+        @Parameter(names = "--topic-c", description = "producer will be created to produce message to this topic",
+                required = true)
+        public List<String> producerTopic = Collections.singletonList("test");
+
+        @Parameter(names = {"-threads", "--num-test-threads"}, description = "Number of test threads")
```

Comment:
This thread is for new transaction and ack the `consumerTopic` message and produce message to `producerTopic ` then commit or abort this transaction. Increasing the number of threads will increase the parallelism of the performance test, thereby increasing the intensity of the stress test.